winum.el

```diff
@@ -340,6 +340,16 @@ WINDOW: if specified, the window of which we want to know the number.
 
 ;; Internal functions ----------------------------------------------------------
 
+(defsubst winum--set-window-vector (window-vector)
+  "Use WINDOW-VECTOR according to the current `winum-scope'."
+  (if (eq winum-scope 'frame-local)
+      (puthash (selected-frame)
+               (cons window-vector
+                     (make-hash-table :size winum--window-count))
+               winum--frames-table)
+    (setq winum--window-vector window-vector)
+    (clrhash winum--numbers-table)))
```

Comment:
Is there any reason why you included the `clrhash` in this function ?

I put it in `winum--update` to reset the window vector and numbers table at every update, but I'm not clear why we should execute it every time we set a new window-vector.